app/models/name/taxonomy.rb

```diff
@@ -83,6 +83,66 @@ def has_eol_data?
     false
   end
 
+  # ----------------------------------------------------------------------------
+
+  # Is the Name (potentially) registrable in a fungal nomenclature repository?
+  # This and #unregistrable are used in the views to determine whether
+  # to display the ICN identifier, links to nomenclature records or searches
+  def registrable?
+    # Almost all Names in MO are potentially registrable, so use a blacklist
+    # instead of a whitelist
+    !unregistrable?
+  end
+
+  # Is name definitely in a fungal nomenclature repository?
+  # (A heuristic that works for almost all cases)
+  def unregistrable?
+    rank == :Group ||
+      rank == :Domain ||
+      unpublished? ||
+      # name includes quote marks, but limit this to below Order in order to
+      # account for things like "Discomycetes", which is registered & quoted
+      /"/ =~ text_name && rank >= :Class ||
+      # Use kingdom: Protozoa as a rough proxy for slime molds
+      # Slime molds, which are Protozoa, are in fungal nomenclature registries.
+      # But most Protozoa are not slime molds and there's no efficient way
+      # for MO to tell the difference. So err on the side of registrability.
+      kingdom.present? && /(Fungi|Protozoa)/ !~ kingdom
+  end
+
+  # Name for which it makes sense to do a fungal nomenclature search
+  # for the stripped monomial, binomial, or combination,
+  # regardless whether name is registrable
+  # For example, Boletus edulis group; it make sense to search for B. edulis
+  def searchable_in_registry?
+    # Use blacklist instead of whitelist; almost all MO names are searchable
+    !unsearchable_in_registry?
+  end
+
+  def unsearchable_in_registry?
+    kingdom.present? && /(Fungi|Protozoa)/ !~ kingdom ||
+      rank == :Domain ||
+      /\bcrypt temp\b/i =~ author&.delete(".")
+  end
+
+  ################
+
+  private
+
+  # Not published in any sense, or not intended as an ICN publication of a name.
+  def unpublished?
+    /\b(nom prov|comb prov|sensu lato|ined)\b/i =~ author&.delete(".")
+  end
+
+  # Kingdom as a string, e.g., "Fungi", or nil if no Kingdom
+  def kingdom
+    parse_classification.find { |rank| rank.first == :Kingdom }&.last
+  end
+
+  public
```

Comment:
Impressive bit of logic!  I'll trust you that it works in most cases.  And for the record I am fine with erring on the side of calling things registrable / searchable even when they aren't.

The other way around could be quite annoying.  That is, omitting a search link where it is perfectly possible to include one, but we just don't think it would work because it's a screwy case.  I'd much rather erroneously include a broken link in some or even all cases instead of omitting valid links for unusual names.

Of course, I could understrand people disagreeing with me, though.  (For example, I can imagine some users complaining "MO is broken!" because it includes a broken link for some screwy name.  Ugh.)